pyuvsim/simsetup.py

```diff
@@ -879,70 +939,45 @@ def initialize_uvdata_from_params(obs_params):
     # Parse time structure
     time_dict = param_dict['time']
     uvparam_dict.update(parse_time_params(time_dict))
-    uvparam_dict['Npols'] = 4
+    uvparam_dict['polarizations'] = np.array([-5, -6, -7, -8])
 
-    # Now make a UVData object with these settings built in.
-    # The syntax below allows for other valid uvdata keywords to be passed
-    #  without explicitly setting them here.
+    # setup object
+    if 'select' in param_dict:
+        uvparam_dict.update(param_dict['select'])
+    if 'object_name' in param_dict:
+        uvparam_dict['object_name'] = param_dict['object_name']
+    uv_obj = setup_uvdata(fill_blts=True, **uvparam_dict)
```

Comment:
I'm not sure I like that `initialize_uvdata_object` calls `parse_telescope_params` and then calls `setup_uvdata`, which itself calls `parse_telescope_params`.